### PR TITLE
Update libp2p-tls to fix issue with secured gRPC certificate

### PR DESCRIPTION
**Pull request by peterargue**
Closes https://github.com/onflow/flow-go/issues/1465

Depends on https://github.com/onflow/flow-go/pull/1692, which updates the min go version to 1.16.

Updates required to pull in this change: https://github.com/libp2p/go-libp2p-tls/pull/100

The `libp2p/go-libp2p-tls` PR updates the certificate generated by the `NewIdentity` method to include subject and issuer fields, making it compliant with the x509 Certificate standard ([RFC3280](https://datatracker.ietf.org/doc/html/rfc3280#section-4.1.2)).

This involves updating various other libp2p libraries, which now require go version 1.16.

### CODE REVIEW

**Comment by smnzhu** [comment]
Looks good, but do you mind waiting for https://github.com/onflow/flow-go/pull/1582? I'm worried that the dependency changes will cause some issues. thanks :)

File: utils/grpcutils/grpc.go [hunk 1]
```diff
@@ -36,7 +36,7 @@ func X509Certificate(privKey crypto.PrivateKey) (*tls.Certificate, error) {
 
 	// extract the TLSConfig from it which will contain the generated x509 certificate
 	// (ignore the public key that is returned - it is the public key of the private key used to generate the ID)
-	libp2pTlsConfig, _ := id.ConfigForAny()
+	libp2pTlsConfig, _ := id.ConfigForPeer("")
```
Comment (vishalchangrani) [inline utils/grpcutils/grpc.go:39]: I thought `ConfigForAny` did the same thing as ConfigForPeer("") , no?

File: utils/grpcutils/grpc_test.go [hunk 2]
```diff
@@ -42,10 +42,15 @@ func TestCertificateGeneration(t *testing.T) {
 	// assert that the public key in the cert matches the test public key
 	require.True(t, expectedKey.Equals(pubKey))
 
-	// assert that the the cert is valid for at least an year starting from now
+	// assert that the cert is valid for at least an year starting from now
 	now := time.Now()
 	require.True(t, now.After(cert.NotBefore))
 	require.True(t, cert.NotAfter.After(now.Add(year)))
+
+	// assert that the cert's subject and issuer fields are set and match (self-signed)
```
Comment (vishalchangrani) [inline utils/grpcutils/grpc_test.go:50]: ❤️

**Review by vishalchangrani — APPROVED** [review]

File: utils/grpcutils/grpc.go (hunk 1 above)
Comment (peterargue) [reply utils/grpcutils/grpc.go:39]: it did. they dropped it in the latest release

**Review by Kay-Zee — APPROVED** [review]

**Review by smnzhu — APPROVED** [review]

**Comment by peterargue** [comment]
bors merge

**Pull request merged**